lib/rules/prefer-object-has-own.js

```diff
@@ -36,6 +40,26 @@ function hasLeftHandObject(node) {
     return false;
 }
 
+/**
+ * Checks if the given token is `await` Identefier, `of` Itentifier or a keyword token
+ * Inspired from eslint-plugin-unicorn/rules/fix/fix-space-around-keywords.js
+ * @param {Token} token The token to check.
+ * @returns {boolean} `true` if the token is  is `await` Identefier, `of` Itentifier or a keyword token
+ */
+function isProblematicToken(token) {
+    const { type, value } = token;
+
+    if (
+        (type === "Keyword" && /^[a-z]*$/u.test(value)) ||
+
+        // AwaitExpression or ForOfStatement
+        (type === "Identifier" && (value === "of" || value === "await"))) {
```

Comment:
`of` can also be tested

```js
for (const x of{}.hasOwnProperty.call(object, property));
```

If you want it make more sense, convert boolean to string

```js
for (const x of{}.hasOwnProperty.call(object, property).toString());
```

Reply:
Thanks, I added more test cases.